modules/data.land/inst/doanload_NEON_multi_site.R

```diff
@@ -0,0 +1,111 @@
+#' Title generalized for download NEON files of multiSites. and automatically detect the most recent available NEON data and if data is problematic, try next year.
+#'
+#' @param settings a multisite settings object, should work with a single site setting. 
+#' @param main_store_dir the preferred path to store the downloaded NEON files. the log file and VEG_INFO should also be stored in that path
+#'
+#' @return VEG_INFO: a list object containing veg_info, site IDs of each site.
+#' @export log.txt file describing the progress of download
+
+Download_multiSites_NEON <- function(settings, main_store_dir){
+  #get NEON sites table
+  neonsites <- neonstore::neon_sites(api = "https://data.neonscience.org/api/v0", .token = Sys.getenv("NEON_TOKEN"))
+  
+  #output
+  VEG_INFO <- list()
+  
+  #log file
+  log_txt <- c()
+  
+  #starting FOR LOOP
+  for (i in 1:length(settings)) {
+    #match lon and lat
+    tempLat <- as.numeric(settings[[i]]$run$site$lat)
```

Comment:
If you're working with a multisettings, is it guaranteed that every element in the list will be a settings or do you need to add checks that you're not trying to access the SDA tag or something like that